modules/apps/29-fee/keeper/grpc_query_test.go

```diff
@@ -643,16 +643,61 @@ func (suite *KeeperTestSuite) TestQueryCounterpartyPayee() {
 			ctx := suite.chainA.GetContext()
 			res, err := suite.chainA.GetSimApp().IBCFeeKeeper.CounterpartyPayee(ctx, req)
 
-			if tc.expPass {
+			if tc.errMsg == "" {
 				suite.Require().NoError(err)
 				suite.Require().Equal(expCounterpartyPayeeAddr.String(), res.CounterpartyPayee)
 			} else {
-				suite.Require().Error(err)
+				suite.Require().ErrorContains(err, tc.errMsg)
 			}
 		})
 	}
 }
 
+func (suite *KeeperTestSuite) TestQueryFeeEnabledChannelsWithPagination() {
```

Comment:
This was really interesting. 

In this test, that was before a test case of another one, we create 10 extra channel, then get the first 5 from pagination and compare them.  However, when building them, we loop over an integer and create them. 

BUT, the logic that handles pagination paginates them based on their *lexicographical order*.
This means that if our channels are all after 10 (and before 100, so as to have 2 digits), the two orders are equal.

But in our case, depending on how many tests were ran before that, we could have the case in which our channels were 1,2,3,4,5,6,7,8,9,10, but the pagination would return 1, *10*, 3,4,5,6,7,8,9, making the test fail!

Thanks @chatton for helping me debug this!

Reply:
Very interesting! And thank you both for debugging and fixing it!